src/eve/utils.py

```diff
@@ -555,23 +555,24 @@ def as_dict(self) -> Dict[str, T]:
     asdict = as_dict
 
 
-class FrozenNamespace(Namespace[T]):
-    """An immutable version of :class:`Namespace`.
+class ConstantNamespace(Namespace[T]):
```

Comment:
This class is not immutable and therefore shouldn't be hashable. All the hash-related functionality should be moved back to `FrozenNamespace`